CMakeLists.txt

```diff
@@ -273,6 +273,33 @@ endif(BUILD_WITH_AUTODIFF_SUPPORT)
 
 if(BUILD_WITH_CASADI_SUPPORT)
   add_project_dependency(casadi 3.4.5 REQUIRED PKG_CONFIG_REQUIRES "casadi >= 3.4.5")
+
+  # Normalize CasADi link target across versions.
+  # Prefer imported target (casadi::casadi), then legacy target (casadi), then library variables.
+  if(TARGET casadi::casadi)
+    set(CASADI_TARGET casadi::casadi)
+  elseif(TARGET casadi)
+    set(CASADI_TARGET casadi)
+  elseif(DEFINED CASADI_LIBRARIES)
+    set(CASADI_TARGET ${CASADI_LIBRARIES})
+  elseif(DEFINED casadi_LIBRARIES)
+    set(CASADI_TARGET ${casadi_LIBRARIES})
+  else()
+    message(FATAL_ERROR "CasADi found but no usable target or libraries variable exported.")
+  endif()
+
+  # Expose to subdirectories and cache for external use if needed.
+  set(CASADI_TARGET "${CASADI_TARGET}" CACHE STRING "CasADi target to link" FORCE)
+
+  # Optional: propagate include dirs only when not using imported targets.
+  if(NOT TARGET casadi::casadi AND NOT TARGET casadi)
+    if(DEFINED CASADI_INCLUDE_DIRS)
+      set(CASADI_INCLUDE_DIRS_NORMALIZED "${CASADI_INCLUDE_DIRS}" CACHE PATH "CasADi include dirs" FORCE)
+    elseif(DEFINED casadi_INCLUDE_DIRS)
+      set(CASADI_INCLUDE_DIRS_NORMALIZED "${casadi_INCLUDE_DIRS}" CACHE PATH "CasADi include dirs" FORCE)
+    endif()
+  endif()
+
```

Comment:
It's not clear what you want to achieve here. Especially, `CASADI_INCLUDE_DIRS_NORMALIZED` is not used anywhere in the PR.

```suggestion
```